src/inc/msquic.h

```diff
@@ -403,6 +403,9 @@ typedef enum QUIC_SERVER_RESUMPTION_LEVEL {
 #define QUIC_PARAM_CONN_STREAM_SCHEDULING_SCHEME        20  // QUIC_STREAM_SCHEDULING_SCHEME
 #define QUIC_PARAM_CONN_DATAGRAM_RECEIVE_ENABLED        21  // uint8_t (BOOLEAN)
 #define QUIC_PARAM_CONN_DATAGRAM_SEND_ENABLED           22  // uint8_t (BOOLEAN)
+#ifdef QUIC_API_ENABLE_INSECURE_FEATURES
```

Comment:
Is there a particular reason to hide this behind an ifdef? Just like the validation flag, hiding those behind a flag makes writing common libraries much harder, since even if you include what you use, if you get included after the first use of msquic.h, you still fail. I think a much better option if we want to hide these is an `msquic_unsafe.h` header, that can be included separately.

Reply:
We want to make it hard to use this. We don't want people shooting themselves in the foot with this accidentally. But I also don't want to use a separate file for it.

Reply:
I might also end up putting any flags/code to disable server certificate validation under the same flag.

Reply:
Ok. It just makes it extremely difficult to even write tests, because you have to be very careful about include order. Thats why QUIC_TEST_APIS has to be defined in the high level files usually, rather then helper headers. But its not the end of the world, just makes things a bit painful sometimes.

Reply:
Yeah, I prefer it this way for now.